ocaml/idl/datamodel.ml

```diff
@@ -9809,6 +9821,178 @@ module SDN_controller = struct
+         field ~qualifier:StaticRO ~ty:String ~lifecycle "serial" "serial of the usb device" ~default_value:(Some (VString ""));
+         field ~qualifier:StaticRO ~ty:String ~lifecycle "version" "usb device version" ~default_value:(Some (VString ""));
+         field ~qualifier:StaticRO ~ty:String ~lifecycle "description" "usb device descritption" ~default_value:(Some (VString ""));
+         field ~qualifier:RW ~ty:Bool ~lifecycle "passthrough_enabled" "enabled for passthrough" ~default_value:(Some (VBool false));
+         field ~qualifier:RW ~ty:(Map (String,String)) ~lifecycle:[Published, rel_inverness, ""] "other_config" "additional configuration" ~default_value:(Some (VMap []));
+        ]
+      ~messages:
+        [
+          scan;
+        ]
+      ()
+end
+let pusb = PUSB.obj
+
+(** Groups of USBs *)
+
+module USB_group = struct
+  let lifecycle = [Published, rel_inverness, ""]
+
+  let create = call
+      ~name:"create"
+      ~lifecycle
+      ~versioned_params:[
+        {param_type=(String); param_name="name_label"; param_doc=""; param_release=inverness_release; param_default=Some (VString "")};
+        {param_type=(String); param_name="name_description"; param_doc=""; param_release=inverness_release; param_default=Some (VString "")};
+        {param_type=(Map (String, String)); param_name="other_config"; param_doc=""; param_release=inverness_release; param_default=Some (VMap [])}
+      ]
+      ~result:(Ref _usb_group, "")
+      ~allowed_roles:_R_POOL_OP
+      ()
+
+  let destroy = call
+      ~name:"destroy"
+      ~lifecycle
+      ~params:[
+        Ref _usb_group, "self", "The USB group to destroy"
+      ]
+      ~allowed_roles:_R_POOL_OP
+      ()
+
+  let obj =
+    create_obj
+      ~name:_usb_group
+      ~descr:"A group of compatible USBs across the resource pool"
+      ~doccomments:[]
+      ~gen_constructor_destructor:false
+      ~gen_events:true
+      ~in_db:true
+      ~lifecycle
+      ~messages_default_allowed_roles:_R_POOL_OP
+      ~persist:PersistEverything
+      ~in_oss_since:None
+      ~contents:[
+        uid _usb_group ~lifecycle;
+        namespace ~name:"name" ~contents:(names None RW ~lifecycle) ();
+        field ~qualifier:DynamicRO ~ty:(Set (Ref _pusb)) ~lifecycle "PUSBs" "List of pUSBs in the group";
+        field ~qualifier:DynamicRO ~ty:(Set (Ref _vusb)) ~lifecycle "VUSBs" "List of vUSBs using the group";
+        field ~qualifier:RW ~ty:(Map (String,String)) ~lifecycle "other_config" "Additional configuration" ~default_value:(Some (VMap []));
+      ]
+      ~messages:[
+        create;
+        destroy;
+      ]
+      ()
+end
+let usb_group = USB_group.obj
+
+module VUSB = struct
+  let lifecycle = [Published, rel_inverness, ""]
+
+  let vusb_operations =
+    Enum ("vusb_operations",
+        [
+          "attach", "Attempting to attach this VUSB to a VM";
+          "plug", "Attempting to plug this VUSB to a VM";
```

Comment:
Minor point: "Attempting to plug this VUSB _into_ a VM"

Reply:
updated.